### ISSUE TITLE
Display the number of reviews when pressing “Evaluate”

### ISSUE THREAD

**Issue by dae** [ankitects/anki#3123]
> 
> The number of reviews is very easy to miss if optimization/evaluation happens too fast, so it would be nice if it was displayed here.

Originally reported on https://forums.ankiweb.net/t/display-the-number-of-reviews-when-pressing-evaluate/43287

**Comment by vasll** [comment]
_In file `ts\routes\deck-options\FsrsOptions.svelte`_
When FSRS is computed the function `computeWeights()` is called. There the value of `computeWeightsProgress` is assigned in one of the callbacks. Finally, `computeWeightsProgress.reviews` is the number of reviews that's shown only briefly on the screen during FSRS optimization.

The idea would be to create a persistent variable where to store the latest FSRS optimization reviews count in Anki. For simplicity I tested this with a variable and initially set it as `0`:
```typescript
let latestOptimizationReviewsCount = 0  // TODO store it persistently
```
Then in `computeWeights()` modify the second callback like so:
```typescript
(progress) => {
    if (progress.value.case === "computeWeights") {
        computeWeightsProgress = progress.value.value;

        // Update latestOptimizationReviewsCount
        if (latestOptimizationReviewsCount == 0 || computeWeightsProgress.reviews != latestOptimizationReviewsCount) {
            latestOptimizationReviewsCount = computeWeightsProgress.reviews
        }
    }
},
```
To change the dialog in `checkWeights()` first we need to add a this new .ftl message:
```
deck-config-latest-optimization-reviews = 
    { $reviewCount ->
        [0] No previous reviews were found.
       *[other] { $reviewCount } reviews in latest FSRS optimization.
    }
```

And then add it to the alert
```typescript
alert(
    `Log loss: ${resp.logLoss.toFixed(4)}, RMSE(bins): ${(
        resp.rmseBins * 100
    ).toFixed(2)}%. ${tr.deckConfigSmallerIsBetter()}` +
    ` ${tr.deckConfigLatestOptimizationReviews({reviewCount: latestOptimizationReviewsCount})}`,
),
```

The result:
![Screenshot from 2024-04-12 12-04-44](https://github.com/ankitects/anki/assets/67590845/dd580faa-9b24-448d-9601-6822ac7c71a5)

I'm not sure what kind of DB Anki uses as I'm not familiar with the codebase yet, but if everything looks good I just need to store `latestOptimizationReviewsCount` somewhere and I can send the PR.

_To note:_ From my understanding `computeWeightsProgress.reviews` is only stored temporarily during the optimization process. If a user has previously done a FSRS optimization (before this new code is added) `latestOptimizationReviewsCount` will still start at `0`, and will only be updated when the user does a new FSRS optimization

**Comment by dae** [comment]
You can see that ComputeFsrsWeightsResponse in the .proto file returns the number of fsrs items. We could do the same with EvaluateWeightsResponse.

**Comment by vasll** [comment]
I initially didn't look at protobuf, my bad

You mean modifying `EvaluateWeightsResponse` like this?
```protobuf
message EvaluateWeightsResponse {
  float log_loss = 1;
  float rmse_bins = 2;
  uint32 fsrs_items = 3; <- Add this field
}
```

Then at `weights.rs` I'd have to modify `evaluate_weights(...)` to return the count of fsrs_items, but how can I do this? Are you suggesting that I'd have to implement the same logic of `compute_weights(...)` for calculating the fsrs_items?

_From my understanding_ the returned fsrs_items in `compute_weights(...)` is the N reviews from the last FSRS optimization up to the current moment (basically at any point in time, the new FSRS reviews) to be then used for computation, but the intended functionality is to only show the N reviews that were used in the latest FSRS computation/optimization _(which is the number of reviews that's only shown briefly while doing the optimization)_

**Comment by dae** [comment]
I'm afraid I don't have time at the moment to step you through it - if you're not sure how to proceed, this might not be the best issue to take on.

**Comment by Expertium** [comment]
Sorry for saying it so late, but I have another suggestion.
When “Evaluate” shows the metrics, make it like this: “Log loss: 0.5000 → 0.4950, RMSE: 4.5% → 4.3%”. That way users won’t have to write down/memorize the numbers.
